components/server/src/workspace/workspace-starter.ts

```diff
@@ -1467,7 +1469,19 @@ export class WorkspaceStarter {
         spec.setClass(instance.workspaceClass!);
 
         if (workspace.type === "regular") {
-            spec.setTimeout(await userTimeoutPromise);
+            const [defaultTimeout, allowSetTimeout] = await Promise.all([userTimeoutPromise, allowSetTimeoutPromise]);
+            spec.setTimeout(defaultTimeout);
+            if (allowSetTimeout) {
+                if (user.additionalData?.workspaceTimeout) {
+                    try {
+                        let timeout = WorkspaceTimeoutDuration.validate(user.additionalData?.workspaceTimeout);
+                        spec.setTimeout(timeout);
+                    } catch (err) {}
```

Comment:
Do we really want to swallow any exception here?

Reply:
I'd :orange_heart: to see this kind of reasoning backend into comments. :slightly_smiling_face: